### PR TITLE
Group 1

### PR DESCRIPTION
**Pull request by johanneskross**

### CODE REVIEW

File: src/main/java/Airport.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+            } // else
+        } //for
+        System.out.println("model:" + maxPlane.getModel());
+        System.out.println("capacity:" + maxPlane.getMaxLoadCapacity());
+    }
+
+    //Constructor
+    public Airport(List<? extends Plane> planes) {
+        this.planes = planes;
```
Comment (nynyg) [inline src/main/java/Airport.java:39]: check if planes is null pointer?

File: src/main/java/Airport.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        this.printMaxTransportMilitaryPlane();
+    }
+
+    private List<? extends Plane> planes;
```
Comment (stefanSchuhbaeck) [inline src/main/java/Airport.java:43]: Move to top of class definition

File: src/main/java/Airport.java (hunk 1 above)
Comment (stefanSchuhbaeck) [reply src/main/java/Airport.java:39]: `public Airport(@NotNull List<? extends Plane> planes)`

File: src/main/java/Airport.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            else {
+
+            } // else
+        } //for
+        System.out.println("model:" + maxPlane.getModel());
+        System.out.println("capacity:" + maxPlane.getMaxLoadCapacity());
+    }
+
+    //Constructor
+    public Airport(List<? extends Plane> planes) {
```
Comment (nynyg) [inline src/main/java/Airport.java:38]: Ctr moved to the top

File: src/main/java/Airport.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import models.MilitaryType;
```
Comment (nynyg) [inline src/main/java/Airport.java:1]: add an intro/comment to the class

File: src/main/java/Airport.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+import Planes.MilitaryPlane;
+import Planes.Plane;
+
+import java.util.*;
```
Comment (nynyg) [inline src/main/java/Airport.java:5]: import "*" is not a good practice

Comment (nynyg) [reply src/main/java/Airport.java:5]: import java.util.ArrayList

File: src/main/java/Airport.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+public class Airport {
+
+    /**
+     * Prints transport military planes
+     */
+    private void printMaxTransportMilitaryPlane() {
```
Comment (nynyg) [inline src/main/java/Airport.java:12]: should be a public method

File: src/main/java/Airport.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        Plane maxPlane = null;
+
+        // print banner
+        System.out.println ("*********************************************");
```
Comment (stefanSchuhbaeck) [inline src/main/java/Airport.java:17]: maybe return String only and let caller decide to print it to stdout

File: src/main/java/Airport.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
+                        maxPlane = plane;
+                    }
+                }
+            } //if
+            else {
+
+            } // else
+        } //for
+        System.out.println("model:" + maxPlane.getModel());
```
Comment (stefanSchuhbaeck) [inline src/main/java/Airport.java:33]: bug: Will be nullPointer if planes list it empty. Add check.

File: src/main/java/Airport.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+        for (Plane plane : planes) {
+            if (plane instanceof MilitaryPlane) {
+                if (((MilitaryPlane) plane).getType() == MilitaryType.TRANSPORT) {
+                    if (maxPlane != null && maxPlane.getMaxLoadCapacity() < plane.getMaxLoadCapacity()) {
+                        maxPlane = plane;
+                    }
+                }
+            } //if
+            else {
```
Comment (nynyg) [inline src/main/java/Airport.java:29]: empty else

File: src/main/java/Airport.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+    private void printMaxTransportMilitaryPlane() {
+        List<MilitaryPlane> transportMilitaryPlanes = new ArrayList<>();
+        Plane maxPlane = null;
+
+        // print banner
+        System.out.println ("*********************************************");
+        System.out.println ("***** Largest Transport Military Plane ******");
+        System.out.println ("*********************************************");
+
+        for (Plane plane : planes) {
```
Comment (nynyg) [inline src/main/java/Airport.java:21]: comment on what the for-loop is doing (find the place with largest cap.).

**Pull request closed without merging**